gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -336,8 +336,8 @@ public QueueProcessor(BlockingQueue queue) {
     public void run() {
       log.info("Starting queue processing.. " + Thread.currentThread().getName());
       KafkaConsumerRecord record = null;
-      try {
-        while (true) {
+      while (!shutdownRequested) {
```

Comment:
The shutdownRequested flag is set to true before calling `ExecutorsUtils.shutdownExecutorService(..,  5000, TimeUnit.MILLISECONDS);` on shutdown i.e. currently, the executor waits for 5 seconds on shutdown to process any pending messages, however, this change would will stop processing messages immediately.

If we want to continue/retry processing on all exceptions, this can be changed to `while (!Thread.currentThread().isInterrupted()) { `

Reply:
Makes sense, reverted the change since we are returning from the catch statement upon getting interrupted.